src/appleseed.studio/debug/tests/testwindow.cpp

```diff
@@ -365,19 +374,56 @@ namespace
     }
 }
 
-void TestWindow::slot_on_test_item_check_state_changed(QTreeWidgetItem* item, int column)
+void TestWindow::slot_on_test_item_check_state_changed(QTreeWidgetItem* item, int column) const
 {
     update_item_check_state(m_ui->treewidget_tests, item, column);
     update_checked_tests_label();
 }
 
-void TestWindow::slot_check_all_tests()
+namespace
+{
+    bool do_filter_items(QTreeWidgetItem* item, const QRegExp& regexp)
+    {
+        bool any_children_visible = false;
+
+        for (int i = 0; i < item->childCount(); ++i)
+        {
+            if (do_filter_items(item->child(i), regexp))
+            {
+                any_children_visible = true;
```

Comment:
You can add a `break` here to exit the loop as soon as one visible child is found.